definitions/features/candlepin_database.rb

```diff
@@ -45,22 +46,52 @@ def env_content_ids_with_null_content
 
   private
 
+  # rubocop:disable Metrics/MethodLength
   def load_configuration
     raw_config = File.read(CANDLEPIN_DB_CONFIG)
     full_config = Hash[raw_config.scan(/(^[^#\n][^=]*)=(.*)/)]
     uri_regexp = %r{://(([^/:]*):?([^/]*))/([^?]*)\??(ssl=([^&]*))?}
-    uri = uri_regexp.match(full_config['jpa.config.hibernate.connection.url'])
+    url = full_config['jpa.config.hibernate.connection.url']
+    uri = uri_regexp.match(url)
     @configuration = {
       'username' => full_config['jpa.config.hibernate.connection.username'],
       'password' => full_config['jpa.config.hibernate.connection.password'],
       'database' => uri[4],
       'host' => uri[2],
       'port' => uri[3] || '5432',
-      'ssl' => (uri[6] == 'true'),
+      'ssl' => (fetch_extra_param(url, 'ssl') == 'true'),
+      'sslfactory' => fetch_extra_param(url, 'sslfactory'),
       'driver_class' => full_config['jpa.config.hibernate.connection.driver_class'],
-      'url' => full_config['jpa.config.hibernate.connection.url']
+      'url' => url
     }
   end
+  # rubocop:enable Metrics/MethodLength
+
+  def extend_with_db_options
+    db_options = { '-d' => construct_database_string }
+    if check_option_using_cpdb_help('dbhost')
+      db_options['--dbhost'] = configuration['host']
+      db_options['--dbport'] = configuration['port']
+    end
+    db_options
+  end
+
+  def construct_database_string
+    db_str = configuration['database']
+    extra_opts = []
+    extra_opts << "ssl=#{configuration['ssl']}" if configuration['ssl']
+    extra_opts << "sslfactory=#{configuration['sslfactory']}" if configuration['sslfactory']
+    db_str += "?#{extra_opts.join('&')}" unless extra_opts.empty?
+    db_str
+  end
+
+  def fetch_extra_param(url, key_name)
+    query_string = url.split('?')[1]
+    return nil unless query_string
+    output = /#{key_name}=([^&]*)?/.match(query_string)
+    return nil unless output
```

Comment:
The last two lines can be simplified by this, unless there is a reason I can't think of for checking on `output[0].include(key_name)`:

```ruby
output[1] if output
```

Reply:
Thank you @iNecas! 
Yeah, no reason. 
I think, I need to take a cup of coffee.